clouddrift/analysis.py

```diff
@@ -1159,3 +1169,219 @@ def unpack_ragged(
         rows = [rows]
 
     return [ragged_array[indices[n] : indices[n + 1]] for n in rows]
+
+
+def extract_inertial_from_position(
+    longitude: np.ndarray,
+    latitude: np.ndarray,
+    p: int,
+    delta_time: Optional[float] = 3600.0,
```

Comment:
Because the method is based on wavelet transform and fft, it works only for data evenly distributed in time. I do not think that the method should incorporate a linear interpolation in case of uneven data because I think that should be left to the user to prepare the data as they want/need/can.
 
I would like to offer the flexibility for the method to take timedelta objects for the argument `time_delta` but I do not know how to best deal with this, any advice?

Reply:
In that case, let's just call it `time_step` and document clearly that it must be in units of seconds. Depending on what arithmetic is used inside the function, it's possible that it will support timedelta objects out of the box. We need to test it and find out.